test/Table_Tests/src/Common_Table_Operations/Conversion_Spec.enso

```diff
@@ -284,6 +284,90 @@ spec setup =
+                m2.at "E" . cast Value_Type.Date . value_type . should_equal Value_Type.Date
+                m2.at "F" . cast Value_Type.Time . value_type . should_equal Value_Type.Time
+                m2.at "G" . cast Value_Type.Date_Time . value_type . should_equal Value_Type.Date_Time
+
+            Test.specify "will extract matching values from a mixed column and replace unconvertible ones with Nothing" <|
+                t0 = table_builder [["X", ["a", "b", "c", My_Type.Value 42]]]
+                t0.at "X" . value_type . should_equal Value_Type.Mixed
+                r0 = t0.cast ["X"] Value_Type.Integer
+                w0 = Problems.expect_warning Conversion_Failure r0
+                w0.affected_rows_count . should_equal 4
+                r0.at "X" . to_vector . should_equal [Nothing, Nothing, Nothing, Nothing]
+
+                mixed_values = [1, 2.25, "3", "-45.25", "2.0", "X", Date.new 2020, Date_Time.new 2023 12 15 19 25, Time_Of_Day.new 12 30, My_Type.Value 42, True, Nothing, "True"]
+                t2 = table_builder [["super-mix", mixed_values]]
+                t2.at "super-mix" . value_type . should_equal Value_Type.Mixed
+
+                nulls n = Vector.fill n Nothing
+
+                c1 = t2.cast "super-mix" Value_Type.Integer . at "super-mix"
+                c1.value_type . should_equal Value_Type.Integer
+                # Texts are not parsed, since cast does not do Text->numeric.
+                c1.to_vector . should_equal ([1, 2] + (nulls 8) + [1, Nothing, Nothing])
+                w1 = Problems.expect_warning Conversion_Failure c1
+                w1.affected_rows_count . should_equal 9
+
+                c2 = t2.cast "super-mix" Value_Type.Float . at "super-mix"
+                c2.value_type . should_equal Value_Type.Float
+                c2.to_vector . should_equal ([1.0, 2.25] + (nulls 8) + [1.0, Nothing, Nothing])
+                w2 = Problems.expect_warning Conversion_Failure c2
+                w2.affected_rows_count . should_equal 9
+
+                c3 = t2.cast "super-mix" Value_Type.Char . at "super-mix"
+                c3.value_type . should_equal Value_Type.Char
+                date_time_str = Date_Time.new 2023 12 15 19 25 . to_text
+                c3.to_vector . should_equal ["1.0", "2.25", "3", "-45.25", "2.0", "X", "2020-01-01", date_time_str, "12:30:00", "{{{MY Type [x=42] }}}", "True", Nothing, "True"]
+                Problems.assume_no_problems c3
+
+                if setup.test_selection.fixed_length_text_columns then
+                    c3_2 = t2.cast "super-mix" (Value_Type.Char size=2) . at "super-mix"
+                    c3_2.value_type . should_equal (Value_Type.Char size=2)
+                    c3.to_vector . should_equal ["1.", "2.", "3", "-4", "2.", "X", "20", "20", "12", "{{", "Tr", Nothing, "Tr"]
+                    w3_2 = Problems.expect_warning Conversion_Failure c3_2
+                    w3_2.affected_rows_count . should_equal 10
```

Comment:
cc @jdunkerley 

I'm wondering if the 'trimming' of string representations should be noted as `Conversion_Failure` or should it get some separate error type? What do you think?

Reply:
(my motivation is that we could get a bit clearer error message for this case)

Maybe we should add a second variant to `Conversion_Failure`?

```
type Conversion_Failure
    Unfit_For_Target_Type ... # <- existing one
    Text_Representation_Too_long # <- for text trimming case
```

Reply:
I think we should report a problem -- it's not like other methods, where the user is explicitly requesting trimming/truncation, so it might be a surprise and hidden data loss is bad. Text_Representation_Too_long is a good error to know about.

Reply:
It is reported (unless text->text in which case trimming _is_ expected).

I'm mostly wondering if we should have a separate error type for this case or a single error for all conversion issues.

I guess the 1 error with 2 constructors may be a decent compromise.